### PR TITLE
Modal: Remove Fabric Core dep by moving to local transitions.

### PR DESCRIPTION
**Pull request by micahgodbolt**
#### Pull request checklist

- [X] Addresses an existing issue: #953

### CODE REVIEW

**Comment by micahgodbolt** [comment]
the onDismissed() function is not getting called. Might be an issue of moving to onTimeout vs onAnimationEnd

File: packages/office-ui-fabric-react/src/components/Modal/Modal.scss [hunk 1]
```diff
@@ -7,17 +7,20 @@
 // --------------------------------------------------
 // Modal styles
 
+:export { duration: $ms-duration2 }
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/Modal/Modal.scss]: nice that it actually works. this is quite handy

File: packages/office-ui-fabric-react/src/components/Modal/Modal.tsx [hunk 2]
```diff
@@ -79,20 +93,17 @@ export class Modal extends BaseComponent<IModalProps, IDialogState> {
       titleAriaId,
       subtitleAriaId,
     } = this.props;
-    let { id, isOpen, isAnimatingOpen, isAnimatingClose } = this.state;
+    let { id, isOpen, isVisible } = this.state;
+
+    const modalClassName = css('ms-Dialog', styles.root, this.props.className, {
+      ['is-open ']: isOpen,
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/Modal/Modal.tsx]: nit, remove space now that you don't concatenate

File: packages/office-ui-fabric-react/src/components/Modal/Modal.scss (hunk 1 above)
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Modal/Modal.scss]: Where is this `:export` feature documented? That is crazy useful.

**Comment by micahgodbolt** [comment]
https://github.com/css-modules/css-modules/issues/86#issuecomment-158273243

[https://avatars3.githubusercontent.com/u/881113?v=3&s=400]<https://github.com/css-modules/css-modules/issues/86#issuecomment-158273243>

Access SASS variables from css-module in a React ... - GitHub<https://github.com/css-modules/css-modules/issues/86#issuecomment-158273243>
github.com
I have a React component that is using css-modules. Is there a way for me to get access to the SASS variables used in the css-module? My styles.scss file is using a ...





Sent from Outlook<http://aka.ms/weboutlook>

________________________________

File: packages/office-ui-fabric-react/src/components/Modal/Modal.scss (hunk 1 above)
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/Modal/Modal.scss]: @mikewheaton  I found it yesterday via google, and it showed up in an issue on github. someone complained about it breaking webpack HMR, but I think we reload the page anyway for our build? /cc @dzearing

**Pull request merged**